File: books/perl/src/examples/other/modify_time_is_the_same.pl
use strict;
use warnings;
use feature 'say';
use Time::HiRes qw(sleep time stat);

{
    open(my $fh, '>', 'first.txt');
    $fh->flush;
    close $fh;
}

say "before ", time;
#sleep(0.0006);
say "after  ", time;

{
    open(my $fh, '>', 'second.txt');
    $fh->flush;
    close $fh;
}

my $first = (stat('first.txt'))[9];
my $second = (stat('second.txt'))[9];
say "first  $first";
say "second $second";
say $first == $second ? "same" : "different";

File: books/python-other/src/examples/other/modify_time_is_the_same.py
import os
import time

with open("first.txt", "w") as fh:
    fh.flush()
    pass
print(f"time: {time.time()}")
#time.sleep(0.01)
with open("second.txt", "w") as fh:
    pass
first = os.path.getmtime("first.txt")
second = os.path.getmtime("second.txt")
print(first)
print(second)
print("same" if first == second else "diff")

